network/cert_verifier.go

```diff
@@ -28,16 +32,139 @@ var (
+		flag.StringVar(&tlsOcspDbURL, "tls_ocsp_database_url", "", "OCSP service URL, for database certificates only")
+	}
+	flag.StringVar(&tlsOcspRequired, "tls_ocsp_required", OcspRequiredDenyUnknownStr,
+		fmt.Sprintf("How to treat certificates unknown to OCSP: <%s>", strings.Join(OcspRequiredValuesList, "|")))
+	flag.StringVar(&tlsOcspFromCert, "tls_ocsp_from_cert", OcspFromCertPreferStr,
+		fmt.Sprintf("How to treat OCSP server described in certificate itself: <%s>", strings.Join(OcspFromCertValuesList, "|")))
+	flag.BoolVar(&tlsOcspCheckOnlyLeafCertificate, "tls_ocsp_check_only_leaf_certificate", false,
+		"Put 'true' to check only final/last certificate, or 'false' to check the whole certificate chain using OCSP")
+	flag.StringVar(&tlsCrlURL, "tls_crl_url", "", "URL of the Certificate Revocation List (CRL) to use")
+	if separate_client_db_urls {
+		flag.StringVar(&tlsCrlClientURL, "tls_crl_client_url", "", "URL of the Certificate Revocation List (CRL) to use, for client/connector certificates only")
+		flag.StringVar(&tlsCrlDbURL, "tls_crl_database_url", "", "URL of the Certificate Revocation List (CRL) to use, for database certificates only")
+	}
+	flag.StringVar(&tlsCrlFromCert, "tls_crl_from_cert", CrlFromCertPreferStr,
+		fmt.Sprintf("How to treat CRL URL described in certificate itself: <%s>", strings.Join(CrlFromCertValuesList, "|")))
+	flag.BoolVar(&tlsCrlCheckOnlyLeafCertificate, "tls_crl_check_only_leaf_certificate", false,
+		"Put 'true' to check only final/last certificate, or 'false' to check the whole certificate chain using CRL")
+	flag.UintVar(&tlsCrlCacheSize, "tls_crl_cache_size", CrlDefaultCacheSize, "How many CRLs to cache in memory (use 0 to disable caching)")
+	flag.UintVar(&tlsCrlCacheTime, "tls_crl_cache_time", CrlDisableCacheTime,
+		fmt.Sprintf("How long to keep CRLs cached, in seconds (use 0 to disable caching, maximum: %d s)", CrlCacheTimeMax))
+}
+
 // CertVerifier is a generic certificate verifier
 type CertVerifier interface {
 	// Verify checks whether the certificate is revoked.
 	// The error is returned if:
 	// - the certificate was revoked
-	// - (for OCSP) the certificate is not known by OCSP server and we requested tls_ocsp_required == "yes" or "all"
-	// - (for OCSP) if we were unable to contact OCSP server(s) but we really need the response, tls_ocsp_required == "all"
+	// - (for OCSP) the certificate is not known by OCSP server and we requested tls_ocsp_required == "denyUnknown" or "requireGood"
+	// - (for OCSP) if we were unable to contact OCSP server(s) but we really need the response, tls_ocsp_required == "requireGood"
 	Verify(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error
 }
 
+// NewCertVerifierFromArgs creates a CertVerifier based on passed OCSP and CRL command line flags.
+// Ignores `--tls_{ocsp,crl}_{client,database}_url` flags, only uses `--tls_{ocsp,crl}_url` as URL source.
+func NewCertVerifierFromArgs() (CertVerifier, error) {
```

Comment:
IMO, you can drop `FromArgs` from the names.

`NewThingFromOtherThing` constructors typically take that other thing as an argument, which is not the case here. It is possible to rewrite the code to use a dedicated struct with parsed values instead of keeping them all as global variables, then pass a reference to that struct. But it's not a complexity that it's needed right now.

Reply:
OK, will rename (this and two other `New*CertVerifierFromArgs`).
Maybe we'll have another constructor for `CertVerifier` some day, then we may need to rename constructor funcs to distinguish between them.